core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -87,19 +87,21 @@ class ReplicaFetcherThread(name: String,
   // process fetched data
   def processPartitionData(topicPartition: TopicPartition, fetchOffset: Long, partitionData: PartitionData) {
     val replica = replicaMgr.getReplicaOrException(topicPartition)
+    val partition = replicaMgr.getPartition(topicPartition).get
     val records = partitionData.toRecords
 
     maybeWarnIfOversizedRecords(records, topicPartition)
 
     if (fetchOffset != replica.logEndOffset.messageOffset)
-      throw new RuntimeException("Offset mismatch for partition %s: fetched offset = %d, log end offset = %d.".format(
+      new IllegalStateException("Offset mismatch for partition %s: fetched offset = %d, log end offset = %d.".format(
```

Comment:
We should throw the exception, right?

Reply:
Ah... my bad. Thanks much for catching this.